sdk/typescript/src/signers/signer-with-provider.ts

```diff
@@ -121,20 +139,31 @@ export abstract class SignerWithProvider implements Signer {
   async getTransactionDigest(
     tx: Base64DataBuffer | SignableTransaction
   ): Promise<string> {
-    let txBytes: Base64DataBuffer;
+    let txBytesWithTag: Base64DataBuffer;
     if (tx instanceof Base64DataBuffer || tx.kind === 'bytes') {
-      txBytes =
+      txBytesWithTag =
         tx instanceof Base64DataBuffer ? tx : new Base64DataBuffer(tx.data);
     } else {
-      txBytes = await this.serializer.serializeToBytes(
+      txBytesWithTag = await this.serializer.serializeToBytes(
         await this.getAddress(),
         tx
       );
     }
+    const version = await this.provider.getRpcApiVersion();
+    let dataToSign;
+    if (version?.major == 0 && version?.minor < 18) {
+      dataToSign = txBytesWithTag;
+    } else {
+      let without_tag = txBytesWithTag.getData().slice(TRANSACTION_DATA_TYPE_TAG.length);
+      const intentMessage = new Uint8Array(INTENT_BYTES.length + without_tag.length);
+      intentMessage.set(INTENT_BYTES);
+      intentMessage.set(without_tag, INTENT_BYTES.length);
+      
+      dataToSign = new Base64DataBuffer(intentMessage);
+    }
```

Comment:
since we had removed sig from generating tx digest, this part will be removed when this flag at L145 is removed https://github.com/MystenLabs/sui/pull/6445/files#diff-90a054a3b632795af85582441a61df5784805807e652e8167d61978a198a17faL145